File: platines-webapp/platines-back/src/main/java/fr/asipsante/platines/service/PlatinesTechDataService.java
/*
 * (c) Copyright 2017-2024, ANS. All rights reserved.
 */
package fr.asipsante.platines.service;

/**
 * Service pour publier des données techniques sur platines. Créé pour afficher la version
 * applicative.
 *
 * @author ericdegenetais
 */
public interface PlatinesTechDataService {
  String getWebappVersion();
}

File: platines-webapp/platines-front/src/app/services/platines-techdata.service.ts
///
/// (c) Copyright 2017-2024, ANS. All rights reserved.
///

import { HttpClient } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { Observable, catchError, of } from "rxjs";
import { host } from "../../environments/environment";

/**
 * Cette classe expose les données techniques platines.
 * Class créée pour afficher la version de l'application platines calculée lors du build.
 */
@Injectable({providedIn:'root'})
export class PlatinesTechDataService {
    constructor(
        private http:HttpClient
        ){}

    platinesVersion(): Observable<string> {
        return this.http.get<string>(`${host.API_ENDPOINT}techdata/public/version`).pipe(
            catchError(
                (error:any,caught: Observable<any>) => {
                    console.error("Erreur d'extraction des informations de version ",error);
                    return of('--version inconnue (erreur)--');
                }
            )
        );
    }
}